### DISCUSSION TITLE
Render of image not showing

### DISCUSSION THREAD

**Discussion by profcwalker** [SethRobinson/aitools_client#18] [Q&A] [answered]
Hi,

Question. 
Server is up and connected to ComfyUI.
When I click to Generate - the images come up black with 'generate error'.
Would you be able to help me figure this part out.
I have Model files in the comfyui folder and can render from there. 
Not sure the model files are being seen through unity though?
Is there something else I need to setup or point to?
Thank you!

**Comment by SethRobinson** [comment, marked as answer]
Hmm - ok, this can be for a few reasons.

### Tips to the get ComfyUI UI target working

1.  Sometimes ComfyUI doesn't handle API requests after it's been loaded a while, I suspect it doesn't like changing models or something.  If that happens, AI Tools usually shows a quick "ComfyUI reports the render failed" at the bottom.

The fix is to kill and restart the ComfyUI process.

2.  The ComfyUI API workflow we're sending doesn't match your models/path setup (this is the most likely problem)

Ok, do you have the console of ComfyUI's output running somewhere?  I suspect it's showing an error of some kind about it.

Here's what to do to verify/fix the setup.

* Go to your ComfyUI interface (I think you can just click on the "ComfyUI Server 1" button in AI Tools to open a browser window there)
* In the upper left, click the Workflow->Open.  Choose aitools_client\ComfyUI\workflow\flux_workflow.json  (or one of the other ones)
* Click "Queue" in comfyUI.   Does a picture get generated?  Probably not, it probably can't find one of the required files due to different filenames or subfolder or something.
![image](https://github.com/user-attachments/assets/d8f55543-19d3-4871-9746-2c099d0a9a5b)

* Click the files missing and choose the correct ones.  Do NOT make any other changes, it might break stuff later.  (if there are multiple text boxes for examples, aitools won't know which ones to replace with its prompt)

* Click Queue again in ComfyUI, is the image properly generated?  After it is, you can move to the next step.

This workflow file is not directly used by AI Tools, but the API version of it is.

* Click Workflow->Export (API) And name it "flux_api" or something.  flux_api.json will get downloaded.  Go ahead and save this to your aitools_client\ComfyUI folder.  The files in here (API versions of workflows) are the ones that AI Tools are going to use to talk to ComfyUI with.
* Restart AI Tools
* Make sure the API workflow you made is selected and click Generate.  If it works you should see images made.
![image](https://github.com/user-attachments/assets/d8fe16a6-dabc-43b0-836b-017e9f422af1)

Hopefully this fixes the problem - knowing this process is useful because it would allow you to make different kinds of ComyUI setups (different models, resolutions, etc), just keep in mind AI Tools is looking for inputs->text in the .json definition to replace the prompt, that's about all it's doing.

Let me know if that works.  If not, check out the log.txt AI Tools makes, it might have additional info about what went wrong.

**Comment by profcwalker** [comment]
Thank you for the quick reply and steps.
I've included a screenshot of the current state. At a point where the mat 1 and 2 cant be multiplied. Might be my combination of model types or unet? Any help is much appreciated. Thank you :) 
![image 1](https://github.com/user-attachments/assets/7bf48adc-add6-408c-a45a-2785dd52c6bd)

**Comment by SethRobinson** [comment]
Ah, it looks like it's set to SD3, that should be set to Flux, correct?  (I see SD3 on your dual clip loader window) If you truly do want to use SD1/SD2/SD3, I recommend using A1111, my tools have better integration with that.  (it can load available models, change sizes and settings, ect).   I only recommend using ComfyUI for Flux.

That said, if your nodes/graphs are set correctly, ComfyUI should be able to render the image, I think the problem you're having currently is using SD3 and Flux stuff together which won't work.

**Reply by profcwalker** [reply]
Excellent, that works great. Got it going nicely. Thank you!

Next question. Maybe a separate post?
If I wanted to use Flux txt to video setup, would that be possible to integrate with the current systems in place?

**Comment by SethRobinson** [comment]
I think I would have to add a movie player so we could actually see the movies in the little windows, similar to how the pictures work.  I'm interested in this too, I was experimenting with a video generator in ComfyUI recently, but that just before the flux-to-video stuff was out there.   I'll see about adding a generic "hey, if ComfyUI sends us a movie, let's show it" support as that would be dang useful, could generate a story/choose your own adventure full of little clips or something.  But now we need that movie to audio generator and... yeah, fun times. :)

Also, I actually have a newer version I need to release, it has better LLM support, some models use different keywords and just wouldn't work previously

**Comment by profcwalker** [comment]
Hi Seth,

Sounds awesome. I'm curious, I'm actually on a time crunch right now and wondering if that is a feature that's possible within the week?
And an update on the project sounds excellent.
Happy to help in any way to see things come together with it.

Thank you,
Chad





________________________________
